x/group/server/invariants.go

```diff
@@ -28,95 +31,109 @@ func (s serverImpl) tallyVotesInvariant() sdk.Invariant {
 		}
 		prevCtx, _ := ctx.CacheContext()
 		prevCtx = prevCtx.WithBlockHeight(ctx.BlockHeight() - 1)
-		msg, broken, err := tallyVotesInvariant(ctx, prevCtx, s.proposalTable)
-		if err != nil {
-			panic(err)
-		}
+		msg, broken := tallyVotesInvariant(ctx, prevCtx, s.proposalTable)
 		return sdk.FormatInvariant(group.ModuleName, votesInvariant, msg), broken
 	}
 }
 
 func (s serverImpl) groupTotalWeightInvariant() sdk.Invariant {
 	return func(ctx sdk.Context) (string, bool) {
-		msg, broken, err := groupTotalWeightInvariant(ctx, s.groupTable, s.groupMemberByGroupIndex)
-		if err != nil {
-			panic(err)
-		}
+		msg, broken := groupTotalWeightInvariant(ctx, s.groupTable, s.groupMemberByGroupIndex)
 		return sdk.FormatInvariant(group.ModuleName, weightInvariant, msg), broken
 	}
 }
 
-func tallyVotesInvariant(ctx sdk.Context, prevCtx sdk.Context, proposalTable orm.AutoUInt64Table) (string, bool, error) {
+func (s serverImpl) tallyVotesSumInvariant() sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		msg, broken := tallyVotesSumInvariant(ctx, s.groupTable, s.proposalTable, s.groupMemberTable, s.voteByProposalIndex, s.groupAccountTable)
+		return sdk.FormatInvariant(group.ModuleName, votesSumInvariant, msg), broken
+	}
+}
+
+func tallyVotesInvariant(ctx sdk.Context, prevCtx sdk.Context, proposalTable orm.AutoUInt64Table) (string, bool) {
 
 	var msg string
 	var broken bool
 
 	prevIt, err := proposalTable.PrefixScan(prevCtx, 1, math.MaxUint64)
 	if err != nil {
-		return msg, broken, err
+		msg += fmt.Sprintf("PrefixScan failure on proposal table at block height %d\n", prevCtx.BlockHeight())
+		return msg, broken
 	}
+
 	curIt, err := proposalTable.PrefixScan(ctx, 1, math.MaxUint64)
 	if err != nil {
-		return msg, broken, err
+		msg += fmt.Sprintf("PrefixScan failure on proposal table at block height %d\n", ctx.BlockHeight())
+		return msg, broken
 	}
 
 	var curProposals []*group.Proposal
 	_, err = orm.ReadAll(curIt, &curProposals)
 	if err != nil {
-		return msg, broken, err
+		msg += fmt.Sprintf("error while getting all the proposals at block height %d\n", ctx.BlockHeight())
+		return msg, broken
 	}
 
 	var prevProposals []*group.Proposal
 	_, err = orm.ReadAll(prevIt, &prevProposals)
 	if err != nil {
-		return msg, broken, err
+		msg += fmt.Sprintf("error while getting all the proposals at block height %d\n", prevCtx.BlockHeight())
+		return msg, broken
 	}
 
 	for i := 0; i < len(prevProposals); i++ {
 		if prevProposals[i].ProposalId == curProposals[i].ProposalId {
 			prevYesCount, err := prevProposals[i].VoteState.GetYesCount()
 			if err != nil {
-				return msg, broken, err
+				msg += fmt.Sprintf("error while getting yes votes weight of proposal at block height %d\n", prevCtx.BlockHeight())
```

Comment:
should we also specify the error `err` in the returned message? ditto for all the msg below